tests/bgp/test_ipv6_nlri_over_ipv4.py

```diff
@@ -0,0 +1,194 @@
+    }
+
+    logger.info("DUT BGP Config: {}".format(duthost.shell('show run bgp')['stdout']))
+    logger.info("Neighbor BGP Config: {}".format(
+        nbrhosts[tor1]["host"].shell("show run bgp")['stdout']))
+    logger.info('Setup_info: {}'.format(setup_info))
+
+    yield setup_info
+
+    # restore config to original state
+    config_reload(duthost, wait=60)
+    config_reload(tor_neighbors[tor1], wait=60, is_dut=False)
+
+    # verify sessions are established
+    bgp_facts = duthost.bgp_facts(instance_id=asic_index)['ansible_facts']
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'].lower() not in skip_hosts:
+            logger.debug(v['description'])
+            assert v['state'] == 'established'
+
+
+def test_nlri(setup):
+    # show current adjacancies
+    cmd = "show ipv6 route {}".format(setup['dut_nlri_route'])
+    logger.info("DUT Route from neighbor: {}".format(setup['duthost'].shell(cmd)['stdout']))
+    cmd = "show ipv6 route {}".format(setup['neigh_nlri_route'])
+    logger.info("Neighbor Route from DUT: {}".format(setup['neighhost'].shell(cmd)['stdout']))
+
+    # remove current neighbor adjacancy
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "no neighbor {} peer-group {}" \
+        -c "no neighbor {} peer-group {}"'\
+        .format(setup['namespace'], setup['dut_asn'], setup['neigh_ip_v4'], setup['peer_group_v4'],
+                setup['neigh_ip_v6'], setup['peer_group_v6'])
+    setup['duthost'].shell(cmd, module_ignore_errors=True)
+    logger.info("DUT BGP Config After Neighbor Removal: {}".format(setup['duthost'].shell('show run bgp')['stdout']))
+
+    cmd = 'vtysh -n {} -c "config" -c "router bgp {}" -c "no neighbor {} peer-group {}" \
+        -c "no neighbor {} peer-group {}"'.format(setup['namespace'], setup['neigh_asn'], setup['dut_ip_v4'],
+                                                  setup['peer_group_v4'], setup['dut_ip_v6'], setup['peer_group_v6'])
+    setup['neighhost'].shell(cmd, module_ignore_errors=True)
+    logger.info("Neighbor BGP Config After Neighbor Removal: {}".format(setup['neighhost']
+                                                                        .shell(cmd="show run bgp")['stdout']))
+    logger.info("Neighbor BGP IPv4 Summary: {}".format(setup['neighhost'].shell(cmd="show ip bgp summary")[u'stdout']))
+    logger.info("Neighbor BGP IPv6 Summary: {}".format(setup['neighhost']
+                                                       .shell(cmd="show ipv6 bgp summary")[u'stdout']))
+    time.sleep(30)
+
+    # clear BGP table
+    cmd = 'vtysh -n {} -c "clear ip bgp * soft"'.format(setup['namespace'])
+    setup['duthost'].shell(cmd)
+    cmd = 'vtysh -c "clear ip bgp * soft"'
+    setup['neighhost'].shell(cmd)
+
+    # verify route is no longer shared
+    cmd = "show ipv6 route {}".format(setup['dut_nlri_route'])
+    dut_route_out = setup['duthost'].shell(cmd)['stdout']
+    assert setup['neigh_ip_v6'] not in dut_route_out
```

Comment:
Please use the `pytest_assert`(https://github.com/sonic-net/sonic-mgmt/blob/9b1cdee55bbadfc47a54df0ec501869cace17652/tests/common/helpers/assertions.py#L4)

Reply:
@lolyu I have made this change.